src/ethereum/berlin/vm/instructions/memory.py

```diff
@@ -45,24 +38,18 @@ def mstore(evm: Evm) -> None:
         If `evm.gas_left` is less than
         `3` + gas needed to extend memeory.
     """
-    # convert to Uint as start_position + size_to_extend can overflow.
-    start_position = Uint(pop(evm.stack))
+    # STACK
+    start_position = pop(evm.stack)
     value = pop(evm.stack).to_be_bytes32()
```

Comment:
Should we be doing type conversions in the `STACK` section, or should it be purely `pop`s?

Reply:
I like it. It's basically a type annotation. `x = pop(evm.stack).to_signed()` tells you that the opcode treats `x` as a signed value.